File: Examples/scpi/Python/gen_burst.py
#!/usr/bin/env python3

import argparse

parser = argparse.ArgumentParser(description='SCPI: generate sinusoidal signal.')
parser.add_argument('adr',          type=str, default='127.0.0.1', help='provide IP address or URL')
parser.add_argument('-p', '--port', type=int, default=5000,        help='specify SCPI port (default is 5000)')
parser.add_argument('-b', '--bin',  action="store_true",           help='use binary data transfer instead of the default ASCII')
parser.add_argument('--py',         action="store_true",           help='use PyVISA-py (by default the system visa library is used)')
args = parser.parse_args()

###############################################################################
# connect to the instrument
###############################################################################

import visa

rm = visa.ResourceManager('@py' if args.py else '')
#rm.list_resources()
rp = rm.open_resource('TCPIP::{}::{}::SOCKET'.format(args.adr, args.port), read_termination = '\r\n')

###############################################################################
# SCPI exchange
###############################################################################

buffer_size = 2**14

# set output amplitude and offset
rp.write(":SOURce1:VOLTage:IMMediate:AMPlitude 1")
rp.write(":SOURce1:VOLTage:IMMediate:OFFSet 0")

# specify burst mode, sinusoidal waveform
rp.write(":SOURce1:MODE BURSt")
rp.write(":SOURce1:FUNCtion:SHAPe SINusoid")

# burst half the buffer with then idle for quarter buffer, repeat 4 times
rp.write(":SOURce1:BURSt:DATA:REPetitions 1")
rp.write(":SOURce1:BURSt:DATA:LENgth "   + str(1 * buffer_size // 2))
rp.write(":SOURce1:BURSt:PERiod:LENgth " + str(3 * buffer_size // 4))
rp.write(":SOURce1:BURSt:PERiod:NUMber 4")
#rp.write(":SOURce1:BURSt:PERiod:NUMber INFinity")

# reset and start state machine
rp.write(":SOURce1:RESET")
rp.write(":SOURce1:START")

# enable output
rp.write(":OUTPUT1:STATe ON")

# trigger state machine
rp.write(":SOURce1:TRIGger")

rp.close()

File: Examples/scpi/MATLAB/gen_burst.m
%% Define Red Pitaya as TCP/IP object
clc
close all
% IP= '192.168.101.108';          % IP of your Red Pitaya...
IP= 'rp-f01b63.local';          % rp-MAC.local MAC are the last 6 characters of your Red Pitaya
port = 5000;              % If you are using WiFi then IP is:
RP=tcpip(IP, port);      

fopen(RP);
RP.Terminator = 'CR/LF';

%% The example generates 4 bursts of half a sine every 3/4 of sine period
buffer_size = 2^14;

% set output amplitude and offset
fprintf(RP,'SOURce1:VOLTage:IMMediate:AMPlitude 1');  % Set amplitude of the output signal
fprintf(RP,'SOURce1:VOLTage:IMMediate:OFFSet 0');     % Set offset of the output signal

% specify peridic mode, sinusoidal waveform
fprintf(RP,'SOURce1:MODE BURSt');
fprintf(RP,'SOURce1:BURSt:MODE FINite');
fprintf(RP,'SOURce1:FUNCtion:SHAPe SINusoid');

% burst half the buffer with then idle for quarter buffer, repeat 4 times
fprintf(RP,'SOURce1:BURSt:DATA:REPetitions 1');
fprintf(RP,['SOURce1:BURSt:DATA:LENgth '   , num2str(1 * buffer_size / 2,'%1.5f ')]);
fprintf(RP,['SOURce1:BURSt:PERiod:LENgth ' , num2str(3 * buffer_size / 4,'%1.5f ')]);
fprintf(RP,'SOURce1:BURSt:PERiod:NUMber 4');

% reset and start state machine
fprintf(RP,'SOURce1:RESET');
fprintf(RP,'SOURce1:START');

% enable output
fprintf(RP,'OUTPUT1:STATe ON');

% trigger state machine
fprintf(RP,'SOURce1:TRIGger');

%% Close connection to the Red Pitaya
fclose(RP);

